applications/ShapeOptimizationApplication/python_scripts/analyzer_internal.py

```diff
@@ -27,9 +29,26 @@
 except ImportError:
     convdiff_response_factory = None
 
-
 import time as timer
 
+class IterationScope:
+    def __init__(self, output_folder_name, response_id, iteration_number):
+        if (output_folder_name != None):

```

Comment:
I suggest to always initialize the members, but with None if no folder name is given.
Then you don't need to do the "hasattr" check below but can check for None which is more pythonic i think.

Reply:
done :)